frontend/src/components/TaskList.js

```diff
@@ -1,153 +1,120 @@
 
-            this.addNotification('Task added', 'info');
+            addNotification('Task added');
         } catch (error) {
             console.error(error);
-            this.setState({ isLoggedIn: false });
         }
     };
 
-    handleUpdateTask = async task => {
+    const handleUpdateTask = async task => {
         try {
-            await this.props.updateTask(task);
+            await updateTask(task);
 
-            this.addNotification('Task updated', 'info');
+            addNotification('Task updated');
         } catch (error) {
             console.error(error);
-            this.setState({ isLoggedIn: false });
         }
     };
 
-    handleDeleteTask = async id => {
+    const handleDeleteTask = async id => {
         try {
-            await this.props.deleteTask(id);
+            await deleteTask(id);
 
-            this.addNotification('Task removed', 'error');
+            addNotification('Task removed', 'error');
         } catch (error) {
             console.error(error);
-            this.setState({ isLoggedIn: false });
         }
     };
 
-    closeNotification = async ts => {
-        this.setState(state => {
-            const notifications = state.notifications.filter(task => task.ts !== ts);
+    const addNotification = async (message, variant = 'info', ts = Date.now()) => {
+        const newNotification = { variant, message, ts };
 
-            return {
-                notifications
-            };
-        });
+        setNotifications([...notifications, newNotification]);
     };
 
-    addNotification = async (message = '', variant = 'info') => {
-        this.setState(state => {
-            return {
-                notifications: state.notifications.concat({ variant, message, ts: Date.now() })
-            };
-        });
+    const closeNotification = async ts => {
+        const filteredNotifications = notifications.filter(notification => notification.ts !== ts);
+
+        setNotifications(filteredNotifications);
     };
 
-    render() {
-        const { fetching, notifications, isLoggedIn } = this.state;
-        const { tasks } = this.props;
-
-        const tasksList =
-            !fetching &&
-            tasks.map(task => (
-                <Task
-                    key={task._id}
-                    id={task._id}
-                    name={task.name}
-                    completed={task.completed}
-                    updateTask={this.handleUpdateTask}
-                    deleteTask={this.handleDeleteTask}
-                    addNotification={this.addNotification}
-                />
-            ));
-
-        const notificationsComponents = notifications.map(notification => (
-            <Notification
-                key={notification.ts}
-                variant={notification.variant}
-                message={notification.message}
-                onClose={this.closeNotification}
-                ts={notification.ts}
+    const tasksList =
+        !isFetching &&
```

Comment:
if `isFetching` is true you could display some loader

Reply:
done